src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt

```diff
@@ -0,0 +1,727 @@
+
+      @Test
+      fun `404 when there is no user`() {
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(),
+          username = "",
+        )
+          .isNotFound
+          .expectBody()
+          .jsonPath("userMessage")
+          .isEqualTo("Resource with id [] not found.")
+      }
+
+      @Test
+      fun `OK even if user does not have access to the caseload`() {
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(),
+          username = "IEP_USER",
+        ).isOk
+      }
+    }
+
+    @Nested
+    @DisplayName("when discharge is rejected")
+    inner class Failure {
+      @Test
+      fun `404 when offender not found`() {
+        offenderNo = "Z9999ZZ"
+
+        getOffender(offenderNo).isNotFound
+
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(),
+        ).isNotFound
+      }
+
+      @Test
+      fun `400 when the discharge time is in the future`() {
+        createOffenderBooking()
+
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(dischargeTime = LocalDateTime.now().plusHours(1)),
+        )
+          .isBadRequest
+          .expectBody()
+          .jsonPath("userMessage")
+          .isEqualTo("Transfer cannot be done in the future") // TODO SDIT-549 This should be "movement" not "transfer"
+      }
+
+      @Test
+      fun `400 when the discharge time is before last movement`() {
+        createOffenderBooking()
+
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(dischargeTime = LocalDateTime.now().minusDays(2)),
+        )
+          .isBadRequest
+          .expectBody()
+          .jsonPath("userMessage")
+          .isEqualTo("Movement cannot be before the previous active movement")
+      }
+
+      @Test
+      fun `404 when the hospital is not an agency`() {
+        createOffenderBooking()
+
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(hospitalLocationCode = "ZZZ"),
+        )
+          .isNotFound
+          .expectBody()
+          .jsonPath("userMessage")
+          .isEqualTo("No ZZZ agency found")
+      }
+
+      @Test
+      fun `404 when the hospital agency type is not a hospital`() {
+        createOffenderBooking()
+
+        dischargeToHospital(
+          offenderNo,
+          dischargeRequest(hospitalLocationCode = "MDI"),
+        )
+          .isNotFound
+          .expectBody()
+          .jsonPath("userMessage")
+          .isEqualTo("MOORLAND is not a hospital")
+      }
+    }
+
+    @Nested
+    @DisplayName("when discharge to hospital is a success")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class SideEffects {
+
+      @ParameterizedTest
+      @MethodSource("uk.gov.justice.hmpps.prison.api.resource.impl.OffenderResourceIntTest_dischargeToHospital#getOffenderTypes")
+      fun `should update booking`(offenderType: OffenderType) {
+        createOffender(offenderType)
```

Comment:
There is no equivalent to `@MethodSource` on `@BeforeEach` so the setup has to be done in each test separately.